poc/tests/test_vidpf.py

```diff
@@ -209,36 +209,53 @@ def test_malformed_correction_word_payload(self):
-        (seed_cw, ctrl_cw, w_cw, proof_cw) = public_share[malformed_level]
+        (seed_cw, ctrl_cw, w_cw, proof_cw) = pub[malformed_level]
         malformed = bytearray(proof_cw)
         malformed[0] ^= 1
-        public_share[malformed_level] = (seed_cw, ctrl_cw, w_cw, malformed)
+        pub[malformed_level] = (seed_cw, ctrl_cw, w_cw, malformed)
 
         # The tweak doesn't impact the computation until we reach the level
         # with the malformed correction word.
         for level in range(malformed_level, vidpf.BITS):
-            prefixes = tuple(range(2**level))
-            proofs = []
-            for agg_id in range(2):
-                (_beta_share, _out_share, proof) = vidpf.eval(
-                    agg_id,
-                    public_share,
-                    keys[agg_id],
-                    level,
-                    prefixes,
-                    nonce,
+            prefixes = tuple(range(2 ** level))
+            for prefix in prefixes:
+                valid = vidpf.verify(
+                    vidpf.eval(0, pub, keys[0], level, [prefix], nonce)[2],
+                    vidpf.eval(1, pub, keys[1], level, [prefix], nonce)[2],
                 )
-                proofs.append(proof)
-            self.assertFalse(vidpf.verify(proofs[0], proofs[1]))
+
+                # TODO Figure out what's wrong with the analysis below. If we
+                # uncomment the else block, then this test fails about half of
+                # the time, suggesting our assumptions are incorrect.
+                #
+                # If the prefix is on path, then we expect the proofs to be
+                # different. Exactly one of the aggregators corrects the node
+                # proof: if correction word is not malformed, then the
+                # corrected node proof will equal the node proof computed by
+                # its co-aggregator; but since the correction word is
+                # malformed, they will always compute different node proofs.
+                #
+                # If the prefix is off path, then we expect the proofs to be
+                # equal. Both aggregators will correct the node proof or
+                # neither will, and since they compute the same seed, we expect
+                # them to compute the same node proof.
+                #
+                # NOTE Even if true, this doesn't seem to imply an attack on
+                # robustness, at least not immediately. For privacy, it's
+                # crucial to make sure the client and honest aggregator agree
+                # on the public share: if the attacker can feed the aggregator
+                # a correction word that's been mangled in this way, then it
+                # can use it to distinguish whether a given prefix is on-path.
+                if vidpf.is_prefix(prefix, alpha, level):
+                    self.assertFalse(valid)
+                # else:
+                #     self.assertTrue(valid)
```

Comment:
@jimouris I haven't quite figured out where my analysis is going wrong.

Reply:
I figured out what's going on :)

Reply:
Was the test wrong?